arch/amd64_llvm/assemble_intel.cpp

```diff
@@ -0,0 +1,216 @@
+
+1. Redistributions of source code must retain the above copyright notice, this
+list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+extern "C" {
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <sys/wait.h>
+
+#include <linux/limits.h>
+
+#include "common.h"
+#include "arch.h"
+
+#include "assemble.h"
+}
+
+#include <iomanip>
+#include <iostream>
+
+#include <llvm/MC/MCAsmInfo.h>
+#include <llvm/MC/MCContext.h>
+#include <llvm/MC/MCELFStreamer.h>
+#include <llvm/MC/MCInstrInfo.h>
+#include <llvm/MC/MCObjectFileInfo.h>
+#include <llvm/MC/MCParser/MCAsmParser.h>
+#include <llvm/MC/MCRegisterInfo.h>
+#include <llvm/MC/MCSection.h>
+#include <llvm/MC/MCSectionELF.h>
+#include <llvm/MC/MCSubtargetInfo.h>
+#include <llvm/MC/MCTargetAsmParser.h>
+#include <llvm/MC/MCTargetOptions.h>
+#include <llvm/Support/ELF.h>
+#include <llvm/Support/Host.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/SourceMgr.h>
+#include <llvm/Support/TargetRegistry.h>
+#include <llvm/Support/TargetSelect.h>
+#include <llvm/Support/raw_ostream.h>
+
+extern struct options_t options;
+
+template<typename T>
+static const T* string_offset(
+		const llvm::StringRef& s, size_t offset,
+		size_t size = sizeof(T)) {
+	if (offset + size > s.size()) {
+		return nullptr;
+	}
+	return (const T*)(s.data() + offset);
+}
+
+extern "C"
+const
+size_t assemble(
+		uint8_t *const bytecode,
+		const size_t bytecode_sz,
+		const char *const assembly,
+		const size_t asm_sz)
+{
+	using std::unique_ptr;
+
+	llvm::InitializeNativeTarget();
+	llvm::InitializeNativeTargetAsmParser();
+
+	/* TODO: this is only used for MCObjectFileInfo init.  Why? */
+	const llvm::Triple triple{LLVM_HOST_TRIPLE};
+
+	std::string error;
+	const llvm::Target* target =
+		llvm::TargetRegistry::lookupTarget(LLVM_HOST_TRIPLE, error);
+	if (!target) {
+		std::cerr << "Could not find target: " << error << std::endl;
+		abort();
```

Comment:
Normally I prefer `abort()`, but you're right—that doesn't seem to help any here.  Will update to use `exit(EXIT_FAILURE)` like the rest of the code.